project/Dependencies.scala

```diff
@@ -95,67 +95,104 @@ object Dependencies {
+      val reactiveStreamsTck = "org.reactivestreams" % "reactive-streams-tck" % "1.0.2" % Test // CC0
     }
 
-    object Provided {
-      // TODO remove from "test" config
+    object ProvidedDependencies {
+      // TODO remove from Test config
       // If changed, update akka-docs/build.sbt as well
-      val sigarLoader = "io.kamon" % "sigar-loader" % "1.6.6-rev002" % "optional;provided;test" // ApacheV2
-
-      val activation = "com.sun.activation" % "javax.activation" % "1.2.0" % "provided;test"
-
-      val levelDB = "org.iq80.leveldb" % "leveldb" % "0.10" % "optional;provided" // ApacheV2
-      val levelDBmultiJVM = "org.iq80.leveldb" % "leveldb" % "0.10" % "optional;provided;multi-jvm" // ApacheV2
-      val levelDBNative = "org.fusesource.leveldbjni" % "leveldbjni-all" % "1.8" % "optional;provided" // New BSD
-
-      val junit = Compile.junit % "optional;provided;test"
+      val sigarLoaderVersion = "1.6.6-rev002"
+      val sigarLoader = Seq(
+        "io.kamon" % "sigar-loader" % sigarLoaderVersion % Optional,
+        "io.kamon" % "sigar-loader" % sigarLoaderVersion % Provided,
+        "io.kamon" % "sigar-loader" % sigarLoaderVersion % Test
+      ) // ApacheV2
+
+      val activationVersion = "1.2.0"
+      val activation = Seq(
+        "com.sun.activation" % "javax.activation" % activationVersion % Provided,
+        "com.sun.activation" % "javax.activation" % activationVersion % Test
+      )
 
-      val scalatest = Def.setting { "org.scalatest" %% "scalatest" % scalaTestVersion.value % "optional;provided;test" } // ApacheV2
+      val levelDBVersion = "0.10"
+      val levelDB = Seq(
+        "org.iq80.leveldb" % "leveldb" % levelDBVersion % Optional,
+        "org.iq80.leveldb" % "leveldb" % levelDBVersion % Provided
+      ) // ApacheV2
+
+      val levelDBmultiJVMVersion = "0.10"
+      val levelDBmultiJVM = Seq(
+        "org.iq80.leveldb" % "leveldb" % levelDBmultiJVMVersion % Optional,
+        "org.iq80.leveldb" % "leveldb" % levelDBmultiJVMVersion % Provided,
+        "org.iq80.leveldb" % "leveldb" % levelDBmultiJVMVersion % "multi-jvm"
+      ) // ApacheV2
+
+      val levelDBNativeVersion = "1.8"
+      val levelDBNative = Seq(
+        "org.fusesource.leveldbjni" % "leveldbjni-all" % levelDBNativeVersion % Optional,
+        "org.fusesource.leveldbjni" % "leveldbjni-all" % levelDBNativeVersion % Provided
+      ) // New BSD
+
+      val junit = Seq(
+        Compile.junit % Optional,
+        Compile.junit % Provided,
+        Compile.junit % Test
+      )
 
+      val scalatest = Def.setting {
+        Seq(
+          "org.scalatest" %% "scalatest" % scalaTestVersion.value % Optional,
+          "org.scalatest" %% "scalatest" % scalaTestVersion.value % Provided,
+          "org.scalatest" %% "scalatest" % scalaTestVersion.value % Test
+        )
+      } // ApacheV2
     }
 
   }
 
-  import Compile._
+  import Compile.{ TestDependencies => Test, ProvidedDependencies => Provided, _ }
```

Comment:
I think just use your new names rather than rename in import

Reply:
I thought it was neater this way, since it avoids a big diff in the commit that would occur if we used the new names - felt like it would be easier to understand looking back. If you feel strongly, I can change it as you suggested :)

Reply:
I think I'm with @chbatey on this, though I understand why you did it that way.

Reply:
While in general I like going from stringly-typed to actual objects references, I'm not sure I like any of the alternatives:
* this 'renaming import' is indeed confusing
* renaming Dependencies.Test to something else and using the new name everywhere is not very clear either
* moving all dependencies to directly under `Dependencies` and changing `Test.foo` to `foo % Test` is kind of nice, but then it's no longer obvious which artifacts are compile dependencies and which are test dependencies.

This makes me think perhaps the current stringly-typed approach isn't so bad after all...

Reply:
We don't need to rename `Dependencies.Test` and `Dependencies.Provided`. It's enough to prefix the scope with `sbt`, for instance:

```scala
 val activation = Seq(
  "com.sun.activation" % "javax.activation" % activationVersion % sbt.Provided,
  "com.sun.activation" % "javax.activation" % activationVersion % sbt.Test
)
```